src/main/java/com/teenthofabud/wizard/nandifoods/wms/settings/unit/uom/service/impl/UOMServiceImpl.java

```diff
@@ -361,31 +363,90 @@ public void updateExistingUOMByCode(String code, UOMDtoV2 sourceUOMDto) {
+        List<UnitClassSelfLinkageDtoV2> oldLinkedUOMs = old.getLinkedUOMs().orElse(Collections.emptyList());
+        List<UnitClassSelfLinkageDtoV2> newLinkedUOMs = _new.getLinkedUOMs().orElse(Collections.emptyList());
+        if(oldLinkedUOMs.size() > 0 && newLinkedUOMs.size() == 0) {
+            log.debug("clearing all linkedUOMs");
             target.removeLinkedFromUOMs();
-        } else if(old.getLinkedUOMs().isEmpty() && _new.getLinkedUOMs().isPresent()) {
-            List<UnitClassSelfLinkageDtoV2> linkedUOMDto = _new.getLinkedUOMs().get();
-            List<UnitClassSelfLinkageContract> selfLinkageContracts = new ArrayList<>(linkedUOMDto);
-            selfLink(target, Optional.of(selfLinkageContracts));
-        } /*else if(old.getLinkedUOMs().isPresent() && _new.getLinkedUOMs().isEmpty()) { // _new.getLinkedUOMs() is always derived from already saved UOMEntity
-            List<UnitClassSelfLinkageDtoV2> linkedUOMDto = old.getLinkedUOMs().get();
-            List<UnitClassSelfLinkageContract> selfLinkageContracts = new ArrayList<>(linkedUOMDto);
+        } else if(oldLinkedUOMs.size() == 0 && newLinkedUOMs.size() > 0) {
+            log.debug("adding all linkedUOMs");
+            List<UnitClassSelfLinkageContract> selfLinkageContracts = new ArrayList<>(newLinkedUOMs);
             selfLink(target, Optional.of(selfLinkageContracts));
-            *//*Collections.binarySearch(target.getFromUOMs(), UOMSelfLinkageEntity.builder().to.build())
-            linkedUOMsLeft.stream().map(f -> target.getFromUOMs(). f.getCode().compareTo())
-            // directly assign to UOMEntity because no comparison is needed
-            linkedUOMsLeft.stream().forEach(f -> {
-                Diff dtoDiff = javers.compare(UnitClassSelfLinkageDtoV2.builder().build(), f);
-                dtoDiff.getChangesByType(ValueChange.class).forEach(Errors.rethrow().wrap(p -> {
-                    BeanUtil.pojo.setProperty(target, "", p.getRight());
-                }));
-            });*//*
-
-        } */else if(old.getLinkedUOMs().isPresent() && _new.getLinkedUOMs().isPresent()) {
-            List<UnitClassSelfLinkageDtoV2> oldLinkedUOMDto = old.getLinkedUOMs().get();
-            List<UnitClassSelfLinkageDtoV2> newLinkedUOMDto = _new.getLinkedUOMs().get();
-            Diff diffCollections = javers.compareCollections(oldLinkedUOMDto, newLinkedUOMDto, UnitClassSelfLinkageDtoV2.class);
+
+        }
+        else if(oldLinkedUOMs.size() > 0 && newLinkedUOMs.size() > 0) {
+
+            Diff diffCollections = javers.compareCollections(oldLinkedUOMs, newLinkedUOMs, UnitClassSelfLinkageDtoV2.class);
             log.debug(diffCollections.prettyPrint());
+            diffCollections.getChangesByType(ValueChange.class).forEach(Errors.rethrow().wrap(p -> {
+
+                if(!(p instanceof InitialValueChange || p instanceof TerminalValueChange) ){
+                    if(p.getPropertyName().equals("quantity")) {
```

Comment:
do not check for individual property names, as this logic needs to be dynamic for extending to other types of resources in other sub domains